### PR TITLE
VACMS-5784: Remove weight set on VISN titles items.

### LINKED ISSUE #5784: VISNs with ids over 10 are sorted incorrectly
**Describe the defect**
VISNs are now sorted by alphabetical, but this puts VISN 2 after VISN 10. 

**To Reproduce**
1. Go to https://prod.cms.va.gov/admin/content and look at the Sections filter
2. VISN 2 appears after VISN 10

**Expected behavior**
VISN 2 should appear after VISN 1

**Screenshots**

**Desktop (please complete the following information if relevant, or delete):**

**Additional context**
Add any other context about the problem here. Reach out to the Product Managers to determine if it should be escalated as critical (prevents users from accomplishing their work with no known workaround and needs to be addressed within 2 business days).

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS objective (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team

Please leave only the team that will do this work selected. If you're not sure, it's fine to leave both selected.

- [ ] `Core Application Team`
- [x] `Product Support Team`

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
Closes #5784
**PER SCRUM CONVO: Remove weight set on save logic, and manually set weights for VISN's in ui.
cc: @kevwalsh @swirtSJW** 

## Testing done
Visual

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `Core Application Team`
- [x] `Product Support Team`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change? 
- [ ] Yes, and it's written in issue ____ and queued for publication. 
  - [ ] Merge and ping @ rachel-kauff so she's ready to publish after deployment
- [ ] Yes, but it hasn't yet been written 
  - [ ] Don't merge yet -- ping @ rachel-kauff to prompt her to write and queue content
- [ ] No announcement is needed for this code change. 
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by kevwalsh** [comment]
@ethanteague i wasn't sure how to QA this. On https://pr5787-peybkcjlspzye8lreyamufhbe40m2fe1.ci.cms.va.gov/admin/structure/taxonomy/manage/administration/overview, i set VISN 19 to weight 10, hit save, and it reset itself to 0, which was unexpected. 

also, i saw this long set of errors for the first time, not sure if it's related. 

There are ~240 of these. 
Sections___VA_gov_CMS

**Comment by ethanteague** [comment]
Sorry about that, @kevwalsh ! The method I was using to grab administration vocab was working for me yesterday (on my local)......but obviously isn't now. Not sure if related to the upgrade, or something else. Anyways, swapped out `getVocabularyId` for `bundle` and it seems to be working. Give it a retest when you get a minute (please).

**Comment by indytechcook** [comment]
> Anyways, swapped out getVocabularyId for bundle and it seems to be working.

`getVocabularyId` was deprecated

File: docroot/modules/custom/va_gov_backend/va_gov_backend.module [hunk 1]
```diff
@@ -1932,7 +1932,7 @@ function va_gov_backend_entity_presave(EntityInterface $entity) {
  *   The term on which to perform weight setting operation.
  */
 function _va_gov_backend_set_term_weight(EntityInterface $entity) {
-  if ($entity->bundle() === 'administration') {
+  if ($entity->bundle() === 'administration' && substr($entity->getName(), 0, 4) !== 'VISN') {
     $storage = \Drupal::service('entity_type.manager')->getStorage('taxonomy_term');
     $parents = $storage->loadAllParents($entity->id());
     if (array_key_exists(8, $parents) || array_key_exists(190, $parents)) {
```
Comment (swirtSJW) [inline docroot/modules/custom/va_gov_backend/va_gov_backend.module:1938]: Can we add a comment to clarify what is special about '8' and '190'  or assign them to meaningful variables so we know what they represent?

**Review by swirtSJW — CHANGES_REQUESTED** [review]
I don't think this is the way we want to force a sort exemption.   All we are trying to do it get a "natural" sort rather than pure alpha sort.    Just set the flag on where we are sorting
https://www.php.net/manual/en/function.sort.php
https://www.php.net/manual/en/function.natsort.php

**Comment by ethanteague** [comment]
> I don't think this is the way we want to force a sort exemption. All we are trying to do it get a "natural" sort rather than pure alpha sort. Just set the flag on where we are sorting
> https://www.php.net/manual/en/function.sort.php
> https://www.php.net/manual/en/function.natsort.php

@kevwalsh , whaddya think about @swirtSJW 's approach?

**Comment by kevwalsh** [comment]
> > I don't think this is the way we want to force a sort exemption. All we are trying to do it get a "natural" sort rather than pure alpha sort. Just set the flag on where we are sorting
> > https://www.php.net/manual/en/function.sort.php
> > https://www.php.net/manual/en/function.natsort.php
> 
> @kevwalsh , whaddya think about @swirtSJW 's approach?

Makes sense!

**Comment by ethanteague** [comment]
PER SCRUM CONVO: Remove weight set on save logic, and manually set weights for VISN's in ui.
cc: @kevwalsh @swirtSJW

**Review by swirtSJW — APPROVED** [review]

**Comment by swirtSJW** [comment]
image

**Pull request merged**